### ISSUE TITLE
stdm with dims arg is missing

### ISSUE THREAD

**Issue by denizyuret** [JuliaLang/julia#33280]
https://github.com/JuliaLang/julia/blob/2f426e45c24e0d94c11f503aa696a902f1bb6aed/stdlib/Statistics/src/Statistics.jl#L381 should also have a dims keyword argument.

**Comment by nalimilan** [comment]
You can use the `mean` argument to `std`. `stdm` is basically deprecated, except that we can't do that before 2.0. The docstring should probably point to `std`.

**Comment by denizyuret** [comment]
Currently the mean argument has a problem: https://github.com/JuliaLang/julia/blob/2f426e45c24e0d94c11f503aa696a902f1bb6aed/stdlib/Statistics/src/Statistics.jl#L338 uses `something(mean, Statistics.mean(A, dims=dims))` which ends up calculating the mean whether or not the user provides one (both args are evaluated afaik). Typically people would use the mean option not to calculate the mean twice.

**Comment by nalimilan** [comment]
Indeed, good catch. Fortunately that's easy to fix, just replace this with an `if mean === nothing` block. Would you make a pull request?
